lib/ecto/adapters/myxql/connection.ex

```diff
@@ -222,10 +222,17 @@ if Code.ensure_loaded?(MyXQL) do
     end
 
     @impl true
-    # DB explain opts are deprecated, so they aren't used to build the explain query.
+    # DB explain opts, except format, are deprecated.
     # See Notes at https://dev.mysql.com/doc/refman/5.7/en/explain.html
     def explain_query(conn, query, params, opts) do
-      case query(conn, build_explain_query(query), params, opts) do
+      {explain_opts, opts} = Keyword.split(opts, ~w[format]a)
+      map_format? = {:format, :map} in explain_opts
+
+      case query(conn, build_explain_query(query, explain_opts), params, opts) do
+        {:ok, %MyXQL.Result{rows: rows}} when map_format? ->
+          decoded_result = MyXQL.json_library().decode!(rows)
```

Comment:
```suggestion
          json_library = MyXQL.json_library()
          decoded_result = Enum.map(&json_library.decode!(&1))
```

I think you have to do this. Could you add integration tests similar to `integration_test/pg/explain_test.exs` and would can test to see if the output is as we expect

Reply:
gotcha, will do